modules/xsession.nix

```diff
@@ -33,8 +33,14 @@ in {
           in
             "''${xmonad}/bin/xmonad";
         '';
+        default = ''test -n "$1" && eval "$@"'';
         description = ''
-          Window manager start command.
+          Window manager start command. The default value handles
+          display-manager parameter from <code>de+wm.desktop</code>. If you use
+          home-manager to config windown manager on nixos, you have to handle
+          extra actions, such as
+          <code>services.xserver.displayManager.session.start</code>, by
+          yourself in <code>xsession.initExtra</code>.
```

Comment:
What does the user have to handle specifically when using NixOS?

Reply:
According the document I written, user needs to handle `services.xserver.displayManager.session.start` in nixos to hm if he want to use these functions. Of course, these are simple, but trival. For example:
1. [set background](https://github.com/NixOS/nixpkgs/blob/959c4e82bc0972bae00f1501821812a7f27e1956/nixos/modules/services/x11/desktop-managers/default.nix#L72-L81)
2. [xdg autostart by systemd](https://github.com/NixOS/nixpkgs/pull/127367/files#diff-ff397b92f09714feef981e5fc2ebe39229f93d71e3aee532458174e4ef6153f8R25-R42)
3. [extra session command by nixos](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/x11/window-managers/i3.nix#L60)

In fact, there are few things now. However, a user using both hm and nixos may ignore them and loss several functions, so I add a document for tips.